server/schedule/checker/replica_strategy.go

```diff
@@ -72,12 +72,12 @@ func (s *ReplicaStrategy) SelectStoreToAdd(coLocationStores []*core.StoreInfo, e
 	strictStateFilter := &filter.StoreStateFilter{ActionScope: s.checkerName, MoveRegion: true}
 	targetCandidate := filter.NewCandidates(s.cluster.GetStores()).
 		FilterTarget(s.cluster.GetOpts(), filters...).
-		Sort(isolationComparer).Reverse().Top(isolationComparer). // greater isolation score is better
-		Sort(filter.RegionScoreComparer(s.cluster.GetOpts()))     // less region score is better
+		KeepTheTopStores(isolationComparer, false) // greater isolation score is better
 	if targetCandidate.Len() == 0 {
 		return 0, false
 	}
-	target := targetCandidate.FilterTarget(s.cluster.GetOpts(), strictStateFilter).PickFirst() // the filter does not ignore temp states
+	target := targetCandidate.FilterTarget(s.cluster.GetOpts(), strictStateFilter).
+		PickTheTopStore(filter.RegionScoreComparer(s.cluster.GetOpts()), true) // less region score is better
```

Comment:
The logic here is different from the previous implementation.

Reply:
If you mean the check of `targetCandidate`, it should keep the same with the previous since the second sort won't change the length. The only logic which is changed here is we do `filter` before sort, which may make it faster, but shouldn't change the final result.

Reply:
The RegionScoreComparer should be a long-term state which should be included in the first stage.